lib/src/utils.dart

```diff
@@ -273,6 +273,196 @@ String getLineEnding(String yaml) {
   return windowsNewlines > unixNewlines ? '\r\n' : '\n';
 }
 
+/// Extracts comments for a node that is replaced within a [YamlMap] or
+/// [YamlList] or a top-level [YamlScalar] of the [yaml] string provided.
+///
+/// [currentEndOffset] represents the end offset of [YamlScalar] or [YamlList]
+/// or [YamlMap] being replaced, that is, `end + 1`.
```

Comment:
This is probably better illustrated with an example.

```dart
/// ```dart
/// final yaml = 'my_key: "replaced-value"\n';
/// //                                    ^--- currentEndOffset points to the newline
/// fimal currentEndOffset = 24;
/// ```
```
ofcourse, I might have counted wrong here :rofl: 

Also does this mean that `currentEndOffset` can be greater than the length of the `yaml` document? (`currentEndOffset == yaml.length`, so `yaml[currentEndOffset]` might throw)

**Example**: `k: v` will have `currentEndOffset: 4` (if I'm replacing `v`, right).

Reply:
Sorry, if I'm asking dumb questions here, but it might be worth making some examples.

Reply:
> Also does this mean that `currentEndOffset` can be greater than the length of the `yaml` document? (`currentEndOffset == yaml.length`, so `yaml[currentEndOffset]` might throw)

`currentEndOffset` shouldn't be greater than `yaml.length` when passed in as an argument. May be I should throw if it is?

---
> 
> **Example**: `k: v` will have `currentEndOffset: 4` (if I'm replacing `v`, right).

I think `3` since it's always zero-based.

---

> Sorry, if I'm asking dumb questions here, but it might be worth making some examples.
> 
> And I think we should make sure we're doing it right, if we get he wrong invariants we'll have a lot of bugs to fix :D

Sure, will add examples.